tests/static-analysis/pair.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/**
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+include __DIR__ . '/../../vendor/autoload.php';
+
+use loophp\collection\Collection;
+use loophp\collection\Contract\Collection as CollectionInterface;
+
+/**
+ * @param CollectionInterface<int, int|null> $collection
+ */
+function pair_checkListInt(CollectionInterface $collection): void
+{
+}
+
+/**
+ * @param CollectionInterface<string, string|null> $collection
+ */
+function pair_checkListString(CollectionInterface $collection): void
+{
+}
+
+pair_checkListInt(Collection::fromIterable([1, 2, 3])->pair());
```

Comment:
this example specifically actually gives a PHP warning. I'm not sure if we can do anything to prevent this usage with the type hints, but probably out of scope for this PR

Reply:
I don't have it locally, what's the warning ?

Reply:
```
╭─alex@AGMac ~/dev/collections/collection ‹feature/add-reduce-operation›
╰─$ php tests/static-analysis/pair.php
PHP Warning:  Undefined array key 1 in /Users/alex/dev/collections/collection/src/Operation/Pair.php on line 49

Warning: Undefined array key 1 in /Users/alex/dev/collections/collection/src/Operation/Pair.php on line 49
array(2) {
  [1]=>
  int(2)
  [3]=>
  NULL
}

Reply:
Weird, I don't have it locally. 

/me investigating.

Reply:
Fixed in 995d26dc218d2bef41a54097a252d8c7c61ef93f !